pkg/tools/spire/start.go

```diff
@@ -220,3 +209,23 @@ func writeDefaultConfigFiles(ctx context.Context, spireRoot string) (string, err
 	}
 	return spireSocketName, nil
 }
+
+func execHealthCheck(ctx context.Context, cmdStr string, options ...*exechelper.Option) error {
+	if _, ok := ctx.Deadline(); !ok {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, healthCheckTimeout)
+		defer cancel()
+	}
+
+healthCheck:
```

Comment:
Can we do not use labels? I would suggest just use a for loop
```suggestion:
for {
```